FSW/src/util/comUtil.cpp

```diff
@@ -0,0 +1,268 @@
+/* comUtil.cpp handles NS2 payload commands
+ * Usage:
+ *  Reads incoming commands from serial, adds them to the command queue
+ *  and executes each command. 
+ *      
+ * 
+ * Modules encompassed:
+ *  Command Handling
+ *
+ * Additional files needed for compilation:
+ *  config.hpp
+ *  comUtil.hpp
+ *  
+ */
+
+/* - - - - - - Includes - - - - - - */
+// C++ libraries
+
+// Other libraries
+
+// NS2 headers
+#include "../headers/config.hpp"
+#include "../headers/comUtil.hpp"
+
+
+/* Module Variable Definitions */
+int commandQueue[COMMAND_QUEUE_SIZE] = {};  // initializes as all zeros
+static int queueIndex = 0;                  // next available index of command queue
+static bool isPaused = false;               // indicates if command execution is paused
+
+// TODO: set Serial.setTimeout() in config.hpp
+
+/* - - - - - - Module Driver Functions - - - - - - */
+
+/* - - - - - - commandHandling - - - - - - *
+ * Usage:
+ *  looks for and reads incoming commands via serial, adds them to the command queue, and executes
+ *  each command in the queue
+ * 
+ * Inputs:
+ *  none
+ * 
+ * Outputs:
+ *  None
+ */
+void commandHandling()
+{
+    int newCommand = readCommand(); // read command from serial
+    if (checkMetaCommand(newCommand)) // if new command is a meta command, execute it immediately
+    {
+        executeCommand(newCommand);
+    }
+    else if (newCommand != -1) // if the new command is valid, add it to the queue
+    {
+        queueCommand(newCommand);
+    }
+    executeAllCommands(); // execute all commands in the queue
+}
+
+/* - - - - - - Helper Functions - - - - - - */
+
+/* - - - - - - readCommand - - - - - - *
+ * Usage:
+ *  Reads serial input for an incoming command, checks if it has recieved a valid command,
+ *  and returns the new command
+ * 
+ * Inputs:
+ *  none
+ * 
+ * Outputs:
+ *  a single valid command code, or -1 if no valid code is recieved
+ */
+int readCommand()
+{
+    // read Serial
+    if (Serial.available() > 0) // check if there is serial input
+    {
+        int serialInput = Serial.parseInt(); // read the command
+        if ((serialInput <= static_cast<int>(Command::DO_NOTHING)) && (serialInput > 0)) // check if command is valid
+        {
+            //Serial.println("Command Recieved."); // for debugging
```

Comment:
For future debugging or is this leftover from your own testing? Will you be able to change this code on flight?

Reply:
Side comment: Will there be a telemetry module? Command received counts are a typical industry telemeter.

Reply:
This is leftover from my own testing, but can be uncommented without consequence. I left it in there just in case we want it for debugging later.

Reply:
The closest thing to a telemetry module is probably housekeeping. We can easily keep track of received commands since the last restart, but unless we want to store that data on the flash modules there is no way to keep track of it through a restart of the teensy.